### PR TITLE
Update renaming of columns from yfinance and passing auto_adjust as optional parameter

### PR DESCRIPTION
**Pull request by kuds**
The yFinance library only passes the Adj Close when auto_adjust is set to False. This pull request was created to address Issue #1300 by passing that parameter along with changing how the Pandas Dataframe renames the columns.

### CODE REVIEW

**Comment by zhumingpassional** [comment]
thanks for your  valuable codes.
does it work now after your testing?

File: finrl/meta/preprocessor/yahoodownloader.py [hunk 1]
```diff
@@ -65,16 +69,20 @@ def fetch_data(self, proxy=None) -> pd.DataFrame:
...
-                "tic",
-            ]
+            data_df.rename(
+                columns={
+                    "Date": "date",
+                    "Adj Close": "adjcp",
+                    "Close": "close",
+                    "High": "high",
+                    "Low": "low",
+                    "Volume": "volumne",
```
Comment (dwiuzila) [inline finrl/meta/preprocessor/yahoodownloader.py:79]: There's a typo here: `volumne`

**Comment by kuds** [comment]
@zhumingpassional 

Yes, [Stock_Fundamental.ipynb](https://colab.research.google.com/github/AI4Finance-Foundation/FinRL-Tutorials/blob/master/1-Introduction/Stock_Fundamental.ipynb) and [Stock_NeurIPS2018_1_Data.ipynb](https://colab.research.google.com/github/AI4Finance-Foundation/FinRL-Tutorials/blob/master/1-Introduction/Stock_NeurIPS2018/Stock_NeurIPS2018_1_Data.ipynb) are now the downloading data successful. 

Please note I decided not to raise an exception during the rename operation if one of the renamed columns does not exist in the data frame. I thought that was a more flexible approach just in case yfinance decides to change its columns in the future.

There are additional issues in the Stock_Fundamental notebooks around Stable Baselines3 deprecating OpenAI's version of Gym, which needs to be converted to Farama's version of Gymnasium. That should be its own issue/pull request.

**Comment by kuds** [comment]
The last thing I wanted to add is that the original issue stems from yfinance changing its default parameters for the download method when switching from version 0.2.50 to 0.2.51 ([Compare Report](https://github.com/ranaroussi/yfinance/commit/02785d61a1e3fe2e667850a3d7ab6b3274c19242)). This change should work in both versions 0.2.50 and 0.2.51. 

Google Colab has yfinance pre-installed on 0.2.50, which will sometimes get updated when installing FinRL resulting in the column mismatch issue.

**Pull request merged**